sys/androidmedia/gstamc.c

```diff
@@ -701,29 +701,21 @@ gst_amc_codec_is_feature_supported (GstAmcCodec * codec,
   return is_feature_supported;
 }
 
-
 gboolean
 gst_amc_codec_enable_adaptive_playback (GstAmcCodec * codec,
     GstAmcFormat * format)
 {
-  gint width = 0;
-  gint height = 0;
+  gboolean adaptive_supported;
 
-  if (!gst_amc_codec_is_feature_supported (codec, format, "adaptive-playback")
-      || !gst_amc_format_get_int (format, "width", &width)
-      || !gst_amc_format_get_int (format, "height", &height) || !width
-      || !height) {
-    return FALSE;
+  adaptive_supported = gst_amc_codec_is_feature_supported (codec, format,
+      "adaptive-playback");
+  if (adaptive_supported) {
+    gst_amc_format_set_int (format, "adaptive-playback", 1);
+    /* FIXME: is this really needed or it's set in the previous call ? */
+    // gst_amc_format_set_feature_enabled (format, "adaptive-playback", TRUE);
   }
-
-  gst_amc_format_set_feature_enabled (format, "adaptive-playback", TRUE);
-  /* FIXME: is this really needed or it's set in the previous call ? */
-  gst_amc_format_set_int (format, "adaptive-playback", 1);
-  GST_DEBUG ("Setting max-width = %d max-height = %d", width, height);
-  gst_amc_format_set_int (format, "max-width", width);
```

Comment:
Setting "max-width" and "max-height" was a requirement for adaptive playback to be enabled, that Realtek/Seraphic told us.

https://android.googlesource.com/platform/frameworks/av/+/refs/heads/master/media/libstagefright/ACodec.cpp#1986

Have you confirmed in logs that adaptive playback is really enabled ?
Log message should be something like:
12-30 21:20:18.924   224   988 I OMX_VDEC: Prepare Adaptive playback: enable 1 , max w/h=3840/2160

Reply:
Please read the commit message, copied to this PR description.

Reply:
This a good point. I'm not saying that it does not work (from the POV of the playback)  but when we were checking for adaptive playback the code does not lie, if max-width/height is not set, adaptive playback is not set on the omx decoder as you can see on the code @aslobodeniuk pointed out. Maybe the decoder just work out-of-the-box by sending new SPS/PPS buffers, but the adaptive-playback feature is not completely set. Can you confirm that the same log happens? and if so, what max width/height it prints?

Reply:
Until this PR, we were destroying the decoder with frame size changes. Now
adaptive is actually working, the same decoder keeps working after frame
size changes.
The first omx trace shows adaptive enabled with max 0x0 and afterwards another trace shows the adaptive enabled with the maximum resolution received up to the time. It is able to adapt to frame sizes without need to specify the maximum beforehand. We may expect some
performance penalty when changing to a large frame due to reallocations, but it works.
As explained in the PR and commit message, if we later find some decoder
that fails when not setting the max frame size in advance, we will have to
set it from the maximum supported by the codec, and optionally setting it
to the stream frame size when it is not adaptive / dash.
Regarding to the documentation, it instructs to set the maximum size but I didn't find anywhere what to do if unknown or what happens if not set. The parameters are marked as optional, and the omx decoder doc says it is a hint to allow the decoder to optimize allocations, what leads to think that it should not fail if it is not set. I tried it and indeed it worked fine.

Reply:
BTW, I already studied the code above. It is a test, and it checks that max size is specified for the test. It does not mean it is actually mandatory. Nowhere says what happens if not set, nowhere says it is mandatory, and testing showed that it works without setting it.

Reply:
In this kind of issues I would go beyond the simplish approach of whether it works or not. There is an evidence of code where max-width and max-height is required and if not set [prepareForAdaptivePlayback](https://android.googlesource.com/platform/frameworks/av/+/56ce726) will not be called: this is a fact. Another fact is that the decoder seems to activate correctly adaptive playback if it's not set for the current board. Both are facts and are not mutually exclusive :)
Now, we should think of how much time did we spend fixing this issue, where we had prior context quite fresh, reproducing, analyzing and actually fixing this issue.
Now let's think of a board in which the decoder implementation actually requires to call prepareForAdaptivePlayback and think how much time it would take to reproduce, validate and fix this issue without prior context fresh.
In this kind of scenarios, I would always try to evaluate if the time we can invest now (~3 hours) will pay off the time might loose in the future (1 week).
My 2 cents

Reply:
> There is an evidence of code where max-width and max-height is required and if not set prepareForAdaptivePlayback will not be called: this is a fact. 
I did not find that evidence, may I know from where it comes? Anyway, if there is evidence of that, I'll add immediately the code to set the max from the maximum supported resolution, and a property to set it, so we can use the appropiate maximum when known and the maximum supported otherwise (dash). If the porperty is not set, we'll use the maximum supported.
Regardless, I would like to know the source of such evidence, I was looking for it for some hours without any success. The ACodec source is not an evidence of that.

Reply:
I will create a separate PR for the same issue HS-3027 with the following:
Add max_width and and max_height properties. If these are set, we will use those. If unset (default) or set to -1, we will query the decoder for the maximum supported and use those instead. If set to 0, we will not set max_width and max_height to the decoder.

Setting the values to the maximum supported has the disadvantage of always forcing a maximum allocation, so I advised to do that only if we confirm that there is some decoder that actually requires it, otherwise it would be better to leave it unset. But this raised concerns expressed in the previous comments, and I will add immediately a PR to set the maximum as described above, allowing the options to configure for all possible cases and being the default safe for all.